Auth0/Authentication.swift

```diff
@@ -753,4 +753,44 @@ public extension Authentication {
     func renew(withRefreshToken refreshToken: String, scope: String? = nil) -> Request<Credentials, AuthenticationError> {
         return self.renew(withRefreshToken: refreshToken, scope: scope)
     }
+
+    /**
+     Renew expired user's credentials using user's current credentials
+
+     ```
+     Auth0.renewExpired(credentials, scope: "openid profile offline_access") {
+         guard $0 == nil else { return }
+         print($1)
+     }
+     ```
+
+     or asking the same scopes requested when the refresh token was issued
+
+     ```
+     Auth0.renewExpired(credentials) {
+         guard $0 == nil else { return }
+         print($1)
+     }
+     ```
+
+     - parameter credentials: the client's credentials obtained on auth
+     - parameter scope: scopes to request for the new tokens. By default is nil which will ask for the same ones requested during Auth
+     - parameter callback: callback with the user's credentials or the cause of the error.
+     - important: This method only works for a refresh token obtained after auth with OAuth 2.0 API Authorization.
+     */
+    func renewExpired(_ credentials: Credentials, scope: String? = nil, callback: @escaping (Error?, Credentials?) -> Void) {
```

Comment:
Regarding naming, well `renewExpired` implies only renew when expired, you will get credentials back if they have not expired they are just not renewed as they aren't expired.

Yes, I was conflicted over where to put it.  I put it here as it requires `Authentication` so rather than passing an `Authentication` instance it, but agree the breaking of convention makes it not quite feel right from an API purity point of view.

I ran originally ran through a few scenarios, for example I liked the idea of a static method in `Credentials` or in a `Storage` utility class.